File: kana_data.py
"""
kana_data.py

Contains the kana character set used by the quiz app: both Hiragana and
Katakana, basic gojuon + dakuten/handakuten variants.

Each entry is a tuple: (character, romaji_reading, script_name)
  - character     : the kana character itself
  - romaji_reading: how to pronounce/type it (e.g. "ka")
  - script_name   : either "Hiragana" or "Katakana" -- tells the quiz
                     which script the character belongs to, so the
                     question can state it explicitly.

KANA is a flat list of these tuples covering both scripts. The quiz
picks randomly from this combined list.
"""

# (character, reading, script)
HIRAGANA_ENTRIES = [
    # --- Vowels ---
    ("あ", "a", "Hiragana"), ("い", "i", "Hiragana"), ("う", "u", "Hiragana"),
    ("え", "e", "Hiragana"), ("お", "o", "Hiragana"),

    # --- K ---
    ("か", "ka", "Hiragana"), ("き", "ki", "Hiragana"), ("く", "ku", "Hiragana"),
    ("け", "ke", "Hiragana"), ("こ", "ko", "Hiragana"),
    # --- G (dakuten of K) ---
    ("が", "ga", "Hiragana"), ("ぎ", "gi", "Hiragana"), ("ぐ", "gu", "Hiragana"),
    ("げ", "ge", "Hiragana"), ("ご", "go", "Hiragana"),

    # --- S ---
    ("さ", "sa", "Hiragana"), ("し", "shi", "Hiragana"), ("す", "su", "Hiragana"),
    ("せ", "se", "Hiragana"), ("そ", "so", "Hiragana"),
    # --- Z (dakuten of S) ---
    ("ざ", "za", "Hiragana"), ("じ", "ji", "Hiragana"), ("ず", "zu", "Hiragana"),
    ("ぜ", "ze", "Hiragana"), ("ぞ", "zo", "Hiragana"),

    # --- T ---
    ("た", "ta", "Hiragana"), ("ち", "chi", "Hiragana"), ("つ", "tsu", "Hiragana"),
    ("て", "te", "Hiragana"), ("と", "to", "Hiragana"),
    # --- D (dakuten of T) ---
    ("だ", "da", "Hiragana"), ("ぢ", "ji", "Hiragana"), ("づ", "zu", "Hiragana"),
    ("で", "de", "Hiragana"), ("ど", "do", "Hiragana"),

    # --- N ---
    ("な", "na", "Hiragana"), ("に", "ni", "Hiragana"), ("ぬ", "nu", "Hiragana"),
    ("ね", "ne", "Hiragana"), ("の", "no", "Hiragana"),

    # --- H ---
    ("は", "ha", "Hiragana"), ("ひ", "hi", "Hiragana"), ("ふ", "fu", "Hiragana"),
    ("へ", "he", "Hiragana"), ("ほ", "ho", "Hiragana"),
    # --- B (dakuten of H) ---
    ("ば", "ba", "Hiragana"), ("び", "bi", "Hiragana"), ("ぶ", "bu", "Hiragana"),
    ("べ", "be", "Hiragana"), ("ぼ", "bo", "Hiragana"),
    # --- P (handakuten of H) ---
    ("ぱ", "pa", "Hiragana"), ("ぴ", "pi", "Hiragana"), ("ぷ", "pu", "Hiragana"),
    ("ぺ", "pe", "Hiragana"), ("ぽ", "po", "Hiragana"),

    # --- M ---
    ("ま", "ma", "Hiragana"), ("み", "mi", "Hiragana"), ("む", "mu", "Hiragana"),
    ("め", "me", "Hiragana"), ("も", "mo", "Hiragana"),

    # --- Y ---
    ("や", "ya", "Hiragana"), ("ゆ", "yu", "Hiragana"), ("よ", "yo", "Hiragana"),

    # --- R ---
    ("ら", "ra", "Hiragana"), ("り", "ri", "Hiragana"), ("る", "ru", "Hiragana"),
    ("れ", "re", "Hiragana"), ("ろ", "ro", "Hiragana"),

    # --- W ---
    ("わ", "wa", "Hiragana"), ("を", "wo", "Hiragana"),

    # --- N (single) ---
    ("ん", "n", "Hiragana"),
]

KATAKANA_ENTRIES = [
    # --- Vowels ---
    ("ア", "a", "Katakana"), ("イ", "i", "Katakana"), ("ウ", "u", "Katakana"),
    ("エ", "e", "Katakana"), ("オ", "o", "Katakana"),

    # --- K ---
    ("カ", "ka", "Katakana"), ("キ", "ki", "Katakana"), ("ク", "ku", "Katakana"),
    ("ケ", "ke", "Katakana"), ("コ", "ko", "Katakana"),
    # --- G (dakuten of K) ---
    ("ガ", "ga", "Katakana"), ("ギ", "gi", "Katakana"), ("グ", "gu", "Katakana"),
    ("ゲ", "ge", "Katakana"), ("ゴ", "go", "Katakana"),

    # --- S ---
    ("サ", "sa", "Katakana"), ("シ", "shi", "Katakana"), ("ス", "su", "Katakana"),
    ("セ", "se", "Katakana"), ("ソ", "so", "Katakana"),
    # --- Z (dakuten of S) ---
    ("ザ", "za", "Katakana"), ("ジ", "ji", "Katakana"), ("ズ", "zu", "Katakana"),
    ("ゼ", "ze", "Katakana"), ("ゾ", "zo", "Katakana"),

    # --- T ---
    ("タ", "ta", "Katakana"), ("チ", "chi", "Katakana"), ("ツ", "tsu", "Katakana"),
    ("テ", "te", "Katakana"), ("ト", "to", "Katakana"),
    # --- D (dakuten of T) ---
    ("ダ", "da", "Katakana"), ("ヂ", "ji", "Katakana"), ("ヅ", "zu", "Katakana"),
    ("デ", "de", "Katakana"), ("ド", "do", "Katakana"),

    # --- N ---
    ("ナ", "na", "Katakana"), ("ニ", "ni", "Katakana"), ("ヌ", "nu", "Katakana"),
    ("ネ", "ne", "Katakana"), ("ノ", "no", "Katakana"),

    # --- H ---
    ("ハ", "ha", "Katakana"), ("ヒ", "hi", "Katakana"), ("フ", "fu", "Katakana"),
    ("ヘ", "he", "Katakana"), ("ホ", "ho", "Katakana"),
    # --- B (dakuten of H) ---
    ("バ", "ba", "Katakana"), ("ビ", "bi", "Katakana"), ("ブ", "bu", "Katakana"),
    ("ベ", "be", "Katakana"), ("ボ", "bo", "Katakana"),
    # --- P (handakuten of H) ---
    ("パ", "pa", "Katakana"), ("ピ", "pi", "Katakana"), ("プ", "pu", "Katakana"),
    ("ペ", "pe", "Katakana"), ("ポ", "po", "Katakana"),

    # --- M ---
    ("マ", "ma", "Katakana"), ("ミ", "mi", "Katakana"), ("ム", "mu", "Katakana"),
    ("メ", "me", "Katakana"), ("モ", "mo", "Katakana"),

    # --- Y ---
    ("ヤ", "ya", "Katakana"), ("ユ", "yu", "Katakana"), ("ヨ", "yo", "Katakana"),

    # --- R ---
    ("ラ", "ra", "Katakana"), ("リ", "ri", "Katakana"), ("ル", "ru", "Katakana"),
    ("レ", "re", "Katakana"), ("ロ", "ro", "Katakana"),

    # --- W ---
    ("ワ", "wa", "Katakana"), ("ヲ", "wo", "Katakana"),

    # --- N (single) ---
    ("ン", "n", "Katakana"),
]

# Combined list used by the quiz. Each item: (character, reading, script)
KANA = HIRAGANA_ENTRIES + KATAKANA_ENTRIES

File: kana_data.js
// kana_data.js
// Hiragana and Katakana flashcard data.
// Each entry: { front, answers[], type, label }

const HIRAGANA = [
  // Vowels
  { front:"あ", answers:["a"] },
  { front:"い", answers:["i"] },
  { front:"う", answers:["u"] },
  { front:"え", answers:["e"] },
  { front:"お", answers:["o"] },
  // K
  { front:"か", answers:["ka"] }, { front:"き", answers:["ki"] },
  { front:"く", answers:["ku"] }, { front:"け", answers:["ke"] },
  { front:"こ", answers:["ko"] },
  // G
  { front:"が", answers:["ga"] }, { front:"ぎ", answers:["gi"] },
  { front:"ぐ", answers:["gu"] }, { front:"げ", answers:["ge"] },
  { front:"ご", answers:["go"] },
  // S
  { front:"さ", answers:["sa"] }, { front:"し", answers:["shi"] },
  { front:"す", answers:["su"] }, { front:"せ", answers:["se"] },
  { front:"そ", answers:["so"] },
  // Z
  { front:"ざ", answers:["za"] }, { front:"じ", answers:["ji"] },
  { front:"ず", answers:["zu"] }, { front:"ぜ", answers:["ze"] },
  { front:"ぞ", answers:["zo"] },
  // T
  { front:"た", answers:["ta"] }, { front:"ち", answers:["chi"] },
  { front:"つ", answers:["tsu"] }, { front:"て", answers:["te"] },
  { front:"と", answers:["to"] },
  // D
  { front:"だ", answers:["da"] }, { front:"ぢ", answers:["ji"] },
  { front:"づ", answers:["zu"] }, { front:"で", answers:["de"] },
  { front:"ど", answers:["do"] },
  // N
  { front:"な", answers:["na"] }, { front:"に", answers:["ni"] },
  { front:"ぬ", answers:["nu"] }, { front:"ね", answers:["ne"] },
  { front:"の", answers:["no"] },
  // H
  { front:"は", answers:["ha"] }, { front:"ひ", answers:["hi"] },
  { front:"ふ", answers:["fu"] }, { front:"へ", answers:["he"] },
  { front:"ほ", answers:["ho"] },
  // B
  { front:"ば", answers:["ba"] }, { front:"び", answers:["bi"] },
  { front:"ぶ", answers:["bu"] }, { front:"べ", answers:["be"] },
  { front:"ぼ", answers:["bo"] },
  // P
  { front:"ぱ", answers:["pa"] }, { front:"ぴ", answers:["pi"] },
  { front:"ぷ", answers:["pu"] }, { front:"ぺ", answers:["pe"] },
  { front:"ぽ", answers:["po"] },
  // M
  { front:"ま", answers:["ma"] }, { front:"み", answers:["mi"] },
  { front:"む", answers:["mu"] }, { front:"め", answers:["me"] },
  { front:"も", answers:["mo"] },
  // Y
  { front:"や", answers:["ya"] }, { front:"ゆ", answers:["yu"] },
  { front:"よ", answers:["yo"] },
  // R
  { front:"ら", answers:["ra"] }, { front:"り", answers:["ri"] },
  { front:"る", answers:["ru"] }, { front:"れ", answers:["re"] },
  { front:"ろ", answers:["ro"] },
  // W
  { front:"わ", answers:["wa"] }, { front:"を", answers:["wo"] },
  // N
  { front:"ん", answers:["n"] },
].map(c => ({ ...c, type: "hiragana", label: "reading" }));

const KATAKANA = [
  // Vowels
  { front:"ア", answers:["a"] }, { front:"イ", answers:["i"] },
  { front:"ウ", answers:["u"] }, { front:"エ", answers:["e"] },
  { front:"オ", answers:["o"] },
  // K
  { front:"カ", answers:["ka"] }, { front:"キ", answers:["ki"] },
  { front:"ク", answers:["ku"] }, { front:"ケ", answers:["ke"] },
  { front:"コ", answers:["ko"] },
  // G
  { front:"ガ", answers:["ga"] }, { front:"ギ", answers:["gi"] },
  { front:"グ", answers:["gu"] }, { front:"ゲ", answers:["ge"] },
  { front:"ゴ", answers:["go"] },
  // S
  { front:"サ", answers:["sa"] }, { front:"シ", answers:["shi"] },
  { front:"ス", answers:["su"] }, { front:"セ", answers:["se"] },
  { front:"ソ", answers:["so"] },
  // Z
  { front:"ザ", answers:["za"] }, { front:"ジ", answers:["ji"] },
  { front:"ズ", answers:["zu"] }, { front:"ゼ", answers:["ze"] },
  { front:"ゾ", answers:["zo"] },
  // T
  { front:"タ", answers:["ta"] }, { front:"チ", answers:["chi"] },
  { front:"ツ", answers:["tsu"] }, { front:"テ", answers:["te"] },
  { front:"ト", answers:["to"] },
  // D
  { front:"ダ", answers:["da"] }, { front:"ヂ", answers:["ji"] },
  { front:"ヅ", answers:["zu"] }, { front:"デ", answers:["de"] },
  { front:"ド", answers:["do"] },
  // N
  { front:"ナ", answers:["na"] }, { front:"ニ", answers:["ni"] },
  { front:"ヌ", answers:["nu"] }, { front:"ネ", answers:["ne"] },
  { front:"ノ", answers:["no"] },
  // H
  { front:"ハ", answers:["ha"] }, { front:"ヒ", answers:["hi"] },
  { front:"フ", answers:["fu"] }, { front:"ヘ", answers:["he"] },
  { front:"ホ", answers:["ho"] },
  // B
  { front:"バ", answers:["ba"] }, { front:"ビ", answers:["bi"] },
  { front:"ブ", answers:["bu"] }, { front:"ベ", answers:["be"] },
  { front:"ボ", answers:["bo"] },
  // P
  { front:"パ", answers:["pa"] }, { front:"ピ", answers:["pi"] },
  { front:"プ", answers:["pu"] }, { front:"ペ", answers:["pe"] },
  { front:"ポ", answers:["po"] },
  // M
  { front:"マ", answers:["ma"] }, { front:"ミ", answers:["mi"] },
  { front:"ム", answers:["mu"] }, { front:"メ", answers:["me"] },
  { front:"モ", answers:["mo"] },
  // Y
  { front:"ヤ", answers:["ya"] }, { front:"ユ", answers:["yu"] },
  { front:"ヨ", answers:["yo"] },
  // R
  { front:"ラ", answers:["ra"] }, { front:"リ", answers:["ri"] },
  { front:"ル", answers:["ru"] }, { front:"レ", answers:["re"] },
  { front:"ロ", answers:["ro"] },
  // W
  { front:"ワ", answers:["wa"] }, { front:"ヲ", answers:["wo"] },
  // N
  { front:"ン", answers:["n"] },
].map(c => ({ ...c, type: "katakana", label: "reading" }));

export { HIRAGANA, KATAKANA };